bin/i18n/curriculum_sync_utils.rb

```diff
@@ -0,0 +1,204 @@
+require 'cdo/languages'
+
+require_relative 'curriculum_sync_utils/serializers'
+
+# This module contains all i18n sync logic specific to the content
+# imported into Code Studio from CurriculumBuilder; Lessons, Activities,
+# Resources, Vocabularies, Objectives, etc.
+module CurriculumSyncUtils
+  def self.sync_in
+    puts "Sync in curriculum content"
+    sync_in_serialize
+    # TODO: redaction
+  end
+
+  def self.sync_out
+    puts "Sync out curriculum content"
+    # TODO: restoration
+    sync_out_reorganize
+  end
+
+  # Helper method to get the desired destination subdirectory of the given
+  # script for the sync in. Note this may be a nested directory like "2021/csf"
+  def self.get_script_subdirectory(script)
+    # special-case Hour of Code scripts.
+    return "Hour of Code" if ScriptConstants.script_in_category?(:hoc, script.name)
+
+    # catchall for scripts without courses
+    return 'other' unless script.get_course_version.present?
+
+    # special-case CSF; we want to group all CSF courses together, even though
+    # they all have different course offerings.
+    return File.join(script.get_course_version.key, 'csf') if script.csf?
+
+    # base case
+    return File.join(script.get_course_version.key, script.get_course_version.course_offering.key)
+  end
+
+  # Serialize all curriculum data to the I18N source directory.
+  #
+  # Relies heavily on the custom serializer logic defined in
+  # curriculum_sync_utils/serializers to generate a hash of results with
+  # translator-readable keys, rather than the basic array that is produced by
+  # default.
+  def self.sync_in_serialize
+    Script.all.each do |script|
+      # TODO: what else do we want to consider when deciding what to sync?
+      next unless script.is_migrated?
```

Comment:
The migration from CurriculumBuilder to levelbuilder

Reply:
more specifically, scripts with `is_migrated`:
1. have lesson plans on code studio instead of curriculum builder
2. use the new script serialization/seeding process based on .script_json files rather than .script files